Prevent refinement of the boundary when using kelly refinement

mivaia opened the pull request:
<!-- Please, fill in the description as completely as possible.-->

### Description

<!-- Explain the content of the new feature
       What are the motivations? 
       How is it integrated to the current code? -->

Mesh adaptation was coarsening the mesh on the boundary when using the kelly estimator for the mesh refinement. This lead to "jumpy" results at the boundary.

### Testing

<!-- How has this been tested?
       What are the new test(s) and what feature(s)/parameter(s) does it test?
       Are there changes and/or impacts on current tests, why?
       How did you ensure that the solution works?
       How will you ensure that it will continue to work in the future? -->

No tests yet.

### Documentation

<!-- Does this new feature introduce new simulation parameters? If so, describe them. -->

Modified the mesh.rst file to take this new feature into account. 

### Miscellaneous (will be removed when merged)

<!-- Anything that you would like to add that does not fit into any of the categories above.
       Note that any critical information should be in the categories above.
       Examples:
         Future changes or features that will be added in subsequent pull requests
         Any comments or highlights for the reviewers -->

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Copyright headers are present and up to date
- [x] Lethe documentation is up to date
- [x] New feature has unit test(s) (preferred) or application test(s), and restart files are in the generator folder
- [x] The branch is rebased onto master
- [ ] Changelog (CHANGELOG.md) is up to date
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If the fix is temporary, an issue is opened
- [x] The PR description is cleaned and ready for merge

blaisb reviewed (changes requested):
Great work. If you have an idea of a quick test, it could be relevant to add an application test to check that the feature is good.
Could you also at least add the parameter in the documentation? :) Thank you

mivaia commented:
Some comments/questions

1. I added a test that compares the number of elements with the new feature. Should I do one that exactly compares the mesh?
2. I think I updated the doc, is there somewhere I forgot to modify? 
3. The feature is in the mesh subsection, do you think it would be more interesting to have it in the mesh adaptation section?

blaisb commented:
> Some comments/questions
> 
> 1. I added a test that compares the number of elements with the new feature. Should I do one that exactly compares the mesh?
I think the test you added is fine. The goal is just to show that you obtain a different number of cells with this feature :). Just to have something that tests the feature.

> 2. I think I updated the doc, is there somewhere I forgot to modify?
Nope the doc update is fine.

> 3. The feature is in the mesh subsection, do you think it would be more interesting to have it in the mesh adaptation section?

I think it is better to put it in the mesh adaptation section in my opinion, but it works in both section. However, since it's related to dynamic adaptation, best put it in that section.
